backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/bpartner/relation/BPRelationsService.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * #%L
+ * de.metas.business.rest-api-impl
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.rest_api.bpartner.relation;
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.BPartnerLocationId;
+import de.metas.bpartner.GLN;
+import de.metas.bpartner.api.IBPRelationDAO;
+import de.metas.bpartner.composite.BPartnerComposite;
+import de.metas.bpartner.composite.BPartnerLocation;
+import de.metas.bpartner.service.BPRelation;
+import de.metas.organization.OrgId;
+import de.metas.rest_api.bpartner.impl.bpartnercomposite.JsonServiceFactory;
+import de.metas.rest_api.bpartner.request.JsonRequestBPRelationTarget;
+import de.metas.rest_api.bpartner.response.JsonResponseBPRelationComposite;
+import de.metas.rest_api.bpartner.response.JsonResponseBPRelationItem;
+import de.metas.rest_api.common.JsonBPRelationRole;
+import de.metas.rest_api.common.JsonExternalId;
+import de.metas.rest_api.utils.IdentifierString;
+import de.metas.rest_api.utils.MetasfreshId;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import de.metas.util.lang.ExternalId;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class BPRelationsService
+{
+
+	private final JsonServiceFactory jsonServiceFactory;
+	private final IBPRelationDAO bpRelationDAO = Services.get(IBPRelationDAO.class);
+
+	public BPRelationsService(@NonNull final JsonServiceFactory jsonServiceFactory)
+	{
+		this.jsonServiceFactory = jsonServiceFactory;
+	}
+
+	public JsonResponseBPRelationComposite getRelationsForPartner(@NonNull final OrgId orgId, @NonNull final IdentifierString bpartnerIdentifier)
```

Comment:
technically org is not requiered if the idneitier is a metasfreshId, but i think it's fine to always ask for the org